packages/flutter/lib/src/material/refresh_indicator.dart

```diff
@@ -594,3 +586,170 @@ class RefreshIndicatorState extends State<RefreshIndicator> with TickerProviderS
     );
   }
 }
+
+/// A sliver widget implementing pull-to-refresh content control.
+///
+/// Unlike [RefreshIndicator] the [SliverRefreshIndicator]
+/// beings part of the scrollable and actively occupies scrollable space.
+///
+/// When inserted as the first sliver in a scroll view or behind other slivers
+/// that still lets the scrollable overscroll in front of this sliver (such as
+/// the [SliverAppBar], this widget will:
+///
+///  * Let the user draw inside the overscrolled area via the passed in [builder].
+///  * Trigger the provided [onRefresh] function when overscrolled far enough to
+///    pass [refreshTriggerPullDistance].
+///  * Continue to hold [refreshIndicatorExtent] amount of space for the [builder]
+///    to keep drawing inside of as the [Future] returned by [onRefresh] processes.
+///  * Scroll away once the [onRefresh] [Future] completes.
+///
+/// The [builder] function will be informed of the current [RefreshIndicatorMode]
+/// when invoking it, except in the [RefreshIndicatorMode.inactive] state when
+/// no space is available and nothing needs to be built. The [builder] function
+/// will otherwise be continuously invoked as the amount of space available
+/// changes from overscroll, as the sliver scrolls away after the [onRefresh]
+/// task is done, etc.
+///
+/// Only one refresh can be triggered until the previous refresh has completed
+/// and the indicator sliver has retracted at least 90% of the way back.
+///
+/// Can only be used in downward-scrolling vertical lists that overscrolls. In
+/// other words, refreshes can't be triggered with [Scrollable]s using
+/// [ClampingScrollPhysics] which is the default on Android. To allow overscroll
+/// on Android, use an overscrolling physics such as [BouncingScrollPhysics].
+/// This can be done via:
+///
+///  * Providing a [BouncingScrollPhysics] (possibly in combination with a
+///    [AlwaysScrollableScrollPhysics]) while constructing the scrollable.
+///  * By inserting a [ScrollConfiguration] with [BouncingScrollPhysics] above
+///    the scrollable.
+///  * By using [CupertinoApp], which always uses a [ScrollConfiguration]
+///    with [BouncingScrollPhysics] regardless of platform.
+///
+/// In a typical application, this sliver should be inserted between the app bar
+/// sliver such as [SliverAppBar] and your main scrollable
+/// content's sliver like: [CustomScrollView].
+///
+/// {@tool dartpad}
+/// When the user scrolls past [refreshTriggerPullDistance],
+/// this sample shows material pull to refresh indicator for 1 second and
+/// adds a new item to the top of the list view.
+///
+/// ** See code in examples/api/lib/material/refresh_indicator/sliver_refresh_control.0.dart **
+/// {@end-tool}
```

Comment:
This is causing some of the failures as well. Can you check that the sample code follows the same pattern as other samples? It is throwing an error.